app/services/metrc_service/plant/discard.rb

```diff
@@ -2,56 +2,67 @@ module MetrcService
   module Plant
     class Discard < MetrcService::Base
       NOT_SPECIFIED = 'Not Specified'.freeze
+      METHOD_CALL = {
+        immature: :destroy_immature_plants,
+        mature: :destroy_mature_plants
+      }.freeze
 
       def call
-        payload = send("build_#{plant_state}_payload", discard, batch)
-        action = plant_state == 'immature' ? :destroy_plant_batches : :destroy_plants
-
-        call_metrc(action, payload)
-
+        send(METHOD_CALL[plant_state])
         success!
       end
 
       private
 
+      def destroy_immature_plants
+        call_metrc(:destroy_plant_batches, build_immature_payload)
+      end
+
+      def destroy_mature_plants
+        call_metrc(:destroy_plants, build_mature_payload)
+      end
+
       def plant_state
         tracking_method = seeding_unit.item_tracking_method
-        @plant_state ||= [nil, 'none'].include?(tracking_method) ? 'immature' : 'mature'
+        @plant_state ||= [nil, 'none'].include?(tracking_method) ? :immature : :mature
       end
 
       def transaction
         @transaction ||= get_transaction(:discard_batch)
       end
 
       def discard
-        batch.discard(@relationships.dig('action_result', 'data', 'id'))
+        return @obj if @obj
+
+        id = @relationships.dig('action_result', 'data', 'id')&.to_i || @completion_id
+        @obj = batch.discard(id)
+
+        @obj
       end
 
-      def build_immature_payload(discard, batch)
-        quantity = discard.attributes['quantity']&.to_i
-        reason   = reason_note(discard)
+      def build_immature_payload
+        quantity = discard.quantity&.to_i
+        reason   = reason_note
 
-        [
-          {
-            PlantBatch: batch.arbitrary_id,
-            Count: quantity,
-            ReasonNote: reason,
-            ActualDate: discard.attributes['discarded_at']
-          }
-        ]
+        [{
+          PlantBatch: batch_tag,
+          Count: quantity,
+          ReasonNote: reason,
+          ActualDate: discard.discarded_at
```

Comment:
are we sure that `discard.discarded_at` is equivalent to `discard.attributes['discarded_at']` ?

Reply:
Yes.